utils/use-google-login.tsx

```
@@ -0,0 +1,85 @@
+import { GoogleAuthProvider, signInWithPopup, UserCredential } from "firebase/auth";
+import { doc, getDoc, setDoc } from "firebase/firestore";
+
+const provider = new GoogleAuthProvider();
+provider.addScope('profile');
+provider.addScope('email');
+// provider.addScope('https://www.googleapis.com/auth/contacts.readonly');
+// provider.setCustomParameters({
+//     'login_hint': 'user@example.com'
+// });
+
+export const useGoogleLogin = (setLoading: Function, setError: Function) => {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)